controllers/api/v1/search_test.go

```diff
@@ -0,0 +1,318 @@
+					})
+
+					It("returns correct json response", func() {
+						user := FabricateUser("John", "john@example.com", "secret")
+						filePath := AppRootDir() + "/tests/fixtures/shared/test.jpeg"
+						_, body := CreateMultipartFormData(filePath)
+						client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+						oauthToken := FabricateOAuthToken(client, user.Id)
+						expectedResponse := `{
+							"errors": [
+								{
+									"detail": "Please upload the file in CSV format."
+								}
+							]
+						}`
+
+						header := http.Header{
+							"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())},
+							"Content-Type":  {"multipart/form-data; boundary=multipart-boundary"}}
+
+						response := MakeAuthenticatedRequest("POST", "/api/v1/search", header, body, nil)
+						responseBody, err := ioutil.ReadAll(response.Body)
+						if err != nil {
+							Fail("Reading response body failed: " + err.Error())
+						}
+
+						Expect(string(responseBody)).To(MatchJSON(expectedResponse))
+					})
+				})
+
+				Context("given the file size is more than 5 megabytes", func() {
+					It("returns 401 status code", func() {
+						user := FabricateUser("John", "john@example.com", "secret")
+						filePath := AppRootDir() + "/tests/fixtures/shared/big_file.pdf"
+						_, body := CreateMultipartFormData(filePath)
+						client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+						oauthToken := FabricateOAuthToken(client, user.Id)
+
+						header := http.Header{
+							"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())},
+							"Content-Type":  {"multipart/form-data; boundary=multipart-boundary"}}
+
+						response := MakeAuthenticatedRequest("POST", "/api/v1/search", header, body, nil)
+
+						Expect(response.StatusCode).To(Equal(http.StatusUnprocessableEntity))
+					})
+
+					It("returns correct json response", func() {
+						user := FabricateUser("John", "john@example.com", "secret")
+						filePath := AppRootDir() + "/tests/fixtures/shared/big_file.pdf"
+						_, body := CreateMultipartFormData(filePath)
+						client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+						oauthToken := FabricateOAuthToken(client, user.Id)
+						expectedResponse := `{
+							"errors": [
+								{
+									"detail": "File size can't be more than 5 MB."
+								}
+							]
+						}`
+
+						header := http.Header{
+							"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())},
+							"Content-Type":  {"multipart/form-data; boundary=multipart-boundary"}}
+
+						response := MakeAuthenticatedRequest("POST", "/api/v1/search", header, body, nil)
+						responseBody, err := ioutil.ReadAll(response.Body)
+						if err != nil {
+							Fail("Reading response body failed: " + err.Error())
+						}
+
+						Expect(string(responseBody)).To(MatchJSON(expectedResponse))
+					})
+				})
+				Context("given the file keywords are empty", func() {
```

Comment:
English:

```suggestion
				Context("given the file keywords is empty", func() {
```

However, it would make more sense as:

```suggestion
				Context("given the upload file has NO keywords", func() {
```